src/Microsoft.OData.Core/UriParser/Binders/SelectExpandBinder.cs

```diff
@@ -274,22 +274,22 @@ private SelectItem GenerateSelectItem(SelectTermToken tokenIn)
             IEdmTypeReference elementTypeReference = targetElementType.ToTypeReference();
 
             // $compute
-            ComputeClause compute = BindCompute(tokenIn.ComputeOption, targetNavigationSource, elementTypeReference);
+            ComputeClause compute = BindCompute(tokenIn.ComputeOption, null, targetNavigationSource, elementTypeReference);

```

Comment:
1) I can't understand why input "null" in $select.
2) what's the difference between "navigationSource" and "TargetNavigationSource"?

Reply:
And complex select can also use the "this.NavigationSource"? why input "null"?

---
In reply to: [515430616](https://github.com/OData/odata.net/pull/1906#discussion_r515430616) [](ancestors = 515430616)

Reply:
@xuzhg 
Responses below:
1. We only want to add another `$it` Range variable to the stack for the `ExpandTermToken`, not the `SelectTermToken`.
2. `NavigationSource` is the parent entity set, `TargetNavigationSource` is the expanded entity set.

Reply:
@xuzhg I added a fix to `Select`, so we are adding a `$it` RangeVariable for the `SelectTermToken`

Reply:
#Resolved